lib/pbench/server/api/resources/query_apis/metadata_index/namespace_and_rows.py

```diff
@@ -270,33 +270,60 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
                 "path": "/_search/scroll",
                 "kwargs": {
                     "json": {
-                        "scroll": IterationSamplesRows.SCROLL_EXPIRY,
+                        "scroll": SampleValues.SCROLL_EXPIRY,
                         "scroll_id": scroll_id,
                     }
                 },
             }
 
         # Retrieve the ES indices that belong to this run_id
-        indices = self.get_index(dataset, "result-data-sample")
+        indices = self.get_index(dataset, document_index)
         if not indices:
             self.logger.debug(
-                f"Found no indices matching the prefix result-data-sample"
+                f"Found no indices matching the prefix {document_index!r}"
                 f"for a dataset {dataset!r}"
             )
             abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
 
+        try:
+            template = Template.find(document_index)
+        except TemplateNotFound:
+            self.logger.exception(
+                f"Document template {document_index!r} not found in the database."
+            )
+            abort(HTTPStatus.INTERNAL_SERVER_ERROR, message="Mapping not found")
+
+        mappings = {
+            "properties": {
+                key: value
+                for key, value in template.mappings["properties"].items()
+                if key in document["whitelist"]
+            }
+        }
```

Comment:
This is common code between your two APIs, which could be abstracted into the base class.